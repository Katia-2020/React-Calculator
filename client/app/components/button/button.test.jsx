import React from 'react';
import { shallow } from 'enzyme';
import Button from './button';

describe('Button', () => {
  describe('render', () => {
    it('should render correctly', () => {
      const wrapper = shallow(<Button
        type="function"
        value="clear"
        display="C"
        theme="dark"
        size="big"
      />);

      expect(wrapper).toMatchSnapshot();
    });

    describe('when I am not passing the size prop', () => {
      it('should render correctly', () => {
        const wrapper = shallow(<Button
          type="function"
          value="clear"
          display="C"
          theme="dark"
        />);

        expect(wrapper).toMatchSnapshot();
      });

      it('should add size "small" prop', () => {
        const wrapper = shallow(<Button
          type="function"
          value="clear"
          display="C"
          theme="dark"
        />);

        const received = wrapper.instance().props.size;
        const expected = 'small';

        expect(received).toEqual(expected);
      });
    });
  });
});
