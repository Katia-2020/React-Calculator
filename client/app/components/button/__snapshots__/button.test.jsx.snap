// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render should render correctly 1`] = `
<div
  className="button button--color-dark button--size-big button--value-clear"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
>
  <div
    className="button__text"
  >
    C
  </div>
</div>
`;

exports[`Button render when I am not passing the size prop should render correctly 1`] = `
<div
  className="button button--color-dark button--size-small button--value-clear"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
>
  <div
    className="button__text"
  >
    C
  </div>
</div>
`;
